packages/wallet-sdk/src/sign/SignRequestHandler.ts

```diff
@@ -20,21 +17,15 @@ export class SignRequestHandler {
     this.tryRestoringSignerFromPersistedType();
   }
 
-  async handleRequest(request: RequestArguments, accounts: AddressString[]) {
+  async handleRequest(request: RequestArguments) {
     try {
-      if (request.method === 'eth_requestAccounts') {
-        if (accounts.length > 0) return accounts;
-        return await this.requestAccounts();
-      }
-
       const signer = await this.useSigner();
-      if (accounts.length <= 0) {
-        throw standardErrors.provider.unauthorized(
-          "Must call 'eth_requestAccounts' before other methods"
-        );
+      if (request.method === 'eth_requestAccounts') {
+        const accounts = await signer.handshake();
+        this.updateListener.onConnect();
+        return accounts;
       }
-
-      return await signer.request(request);
+      return signer.request(request);
```

Comment:
we might need to await here, otherwise the error won't get caught here, because we returned promise object and ended here already
```suggestion
      return await signer.request(request);
```